clients/drcachesim/tracer/instru_offline.cpp

```diff
@@ -733,8 +740,16 @@ offline_instru_t::label_marks_elidable(instr_t *instr, OUT int *opnd_index,
 
 void
 offline_instru_t::identify_elidable_addresses(void *drcontext, instrlist_t *ilist,
-                                              int version)
-{
+                                              int version, bool has_scatter_gather_instr)
+{
+    // XXX: We turn off address elision for bbs containing scatter/gather. As
```

Comment:
How about generalizing: check for emulation labels and disallow inside any emulation region.  Then remove the extra bool parameters.  Any emulation will have this same issue, not just scatter/gather.

For the call from raw2trace: maybe add an explicit check for an (unexpanded) scatter/gather or rep-string in the loop below?  Instead of the caller figuring it out and passing in a bool.

Reply:
Removed the extra boolean param and added a check for emulation labels (for the call from instru_offline), and a check for scatter/gather/repstr (for the call from raw2trace).

I had to add a new loop, because the current one looks only at app instrs, and the emulation labels are non-app. Is it worth avoiding this extra loop during raw2trace, where it would be sufficient to look at the app instrs? I guess raw2trace overhead isn't as important as that of instru_offline, so maybe it's okay to have the extra loop.

Reply:
raw2trace overhead does matter: we had to put effort into the various caching and parallelism schemes.  But this info collected is cached, right?  So this would just be called once per static block (not per execution), and there it shouldn't matter.

Reply:
Yes, it seems to be cached. `identify_elidable_addresses` is called only once per bb in instru_offline, and the `analyze_elidable_addresses` call in raw2trace is made only if instr summary doesn't exist already.